.gitignore

```diff
@@ -5,5 +5,4 @@ dist
 
 # these cause more harm than good
 # when working with contributors
-package-lock.json
```

Comment:
> I had to commit the package-lock.json to convince the CI to install the correct versions. So not sure what to do with this.

I might be wrong about this, but we're not a big fan of introducing lock files (https://github.com/testing-library/dom-testing-library/pull/624).
That's why we're also not using it during our CI pipeline:
https://github.com/testing-library/dom-testing-library/blob/a86c54ccda5242ad8dfc1c70d31980bdbf96af7f/.github/workflows/validate.yml#L49

If you want you can revert this change, and I can give it a look in order to fix this.

Reply:
This originates from @kentcdodds' package starters.

If I remember correctly (but Kent can probably go more in depth here) the main reason is for when different people are adding/removing/updating different (dev)dependencies at the same time.
When merging one PR, the other one could automatically be in conflict with the one just merged, which is a pain to resolve.

Reply:
A good solution (for now) would probably be to pin the versions to an exact number and investigate this further in another PR, so we don't hold back this PR

Reply:
That is what I tried exactly. I pinned the versions but the pinned dependencies also have dependencies and they are not pinned. The version I have locally runs for whatever reason. So my set of dependencies and sub dependencies work together.
I spend quite a while to find a set of overwrites and versions that ran at all. So it might take another while to find out what I have installed vs what the ci is installing :D.

Yes, package-lock conflicts are annoying. But npm can handle merge conflicts nowadays and will resolve them for you if you just run "npm install".

We can also move this discussion into its own issue as you suggested. I think it would be worthwhile to decide that first before putting more time into investigating the exact dependencies we need to overwrite :D